packages/shared/src/server/repositories/observations.ts

```diff
@@ -1165,3 +1166,81 @@ export const getTraceIdsForObservations = async (
     traceId: row.trace_id,
   }));
 };
+
+export const getGenerationsForPostHog = async (
+  projectId: string,
+  minTimestamp: Date,
+  maxTimestamp: Date,
+) => {
+  const query = `
+    SELECT
+      o.name as name,
+      o.start_time as start_time,
+      o.id as id,
+      o.total_cost as total_cost,
+      if(isNull(completion_start_time), NULL, date_diff('milliseconds', start_time, completion_start_time)) as time_to_first_token,
+      o.usage_details['total'] as input_tokens,
+      o.usage_details['output'] as output_tokens,
+      o.cost_details['total'] as total_tokens,
+      o.project_id as project_id,
+      if(isNull(end_time), NULL, date_diff('milliseconds', start_time, end_time) / 1000) as latency,
+      o.provided_model_name as model,
+      o.level as level,
+      o.version as version,
+      t.id as trace_id,
+      t.name as trace_name,
+      t.session_id as trace_session_id,
+      t.user_id as trace_user_id,
+      t.release as trace_release,
+      t.tags as trace_tags,
+      t.metadata['$posthog_session_id'] as posthog_session_id
+    FROM observations o FINAL
+    LEFT JOIN traces t FINAL on o.trace_id = t.id and o.project_id = t.project_id
+    WHERE o.project_id = {projectId: String}
+    AND t.project_id = {projectId: String}
+    AND o.start_time >= {minTimestamp: DateTime64(3)}
+    AND o.start_time <= {maxTimestamp: DateTime64(3)}
+    AND t.timestamp >= {minTimestamp: DateTime64(3)} - INTERVAL 7 DAY
+    AND t.timestamp <= {maxTimestamp: DateTime64(3)}
+    AND o.type = 'GENERATION'
+  `;
+
+  const records = await queryClickhouse<Record<string, unknown>>({
+    query,
+    params: {
+      projectId,
+      minTimestamp: convertDateToClickhouseDateTime(minTimestamp),
+      maxTimestamp: convertDateToClickhouseDateTime(maxTimestamp),
+    },
+  });
+
+  const baseUrl = env.NEXTAUTH_URL?.replace("/api/auth", "");
+  return records.map((record) => ({
+    timestamp: record.start_time,
+    langfuse_generation_name: record.name,
+    langfuse_trace_name: record.trace_name,
+    langfuse_url: `${baseUrl}/project/${projectId}/traces/${encodeURIComponent(record.trace_id as string)}?observation=${encodeURIComponent(record.id as string)}`,
+    langfuse_id: record.id,
+    langfuse_cost_usd: record.total_cost,
+    langfuse_input_units: record.input_tokens,
+    langfuse_output_units: record.output_tokens,
+    langfuse_total_units: record.total_tokens,
+    langfuse_session_id: record.trace_session_id,
+    langfuse_project_id: projectId,
+    langfuse_user_id: record.trace_user_id || "langfuse_unknown_user",
+    langfuse_latency: record.latency,
+    langfuse_time_to_first_token: record.time_to_first_token,
+    langfuse_release: record.trace_release,
+    langfuse_version: record.version,
+    langfuse_model: record.model,
+    langfuse_level: record.level,
+    langfuse_tags: record.trace_tags,
+    langfuse_integration_version: "1.0.0",
```

Comment:
@marcklingen I updated the integration version here. I'm not sure how we want to handle semantic versioning here, but it feels good to just set it to 1.0.0 now that we moved it into the codebase.

Reply:
No strong feeling about it either. Might be nice to keep it here as it allows us to version events separately. That way it becomes immediately obvious that a version should be bumped. As it's additional effort to change and there are no strong opinions, I'll keep as is.

Reply:
naming here suggests "integration" versioning

i think changing to "langfuse_event_version" would be good here as well, no strong opinion

Reply:
Adjusted in https://github.com/langfuse/langfuse/pull/4685/commits/254799ec58f793f082e30ee73d754146f08ec28d. I like it!